bin/hdfcoinc/pycbc_multiifo_coinc_findtrigs

```diff
@@ -0,0 +1,353 @@
+
+    logging.info('Calculating Multi-Detector Combined Statistic')
+
+    sdswithids={}
+    for ifo in sds:
+        sdswithids[ifo]=sds[ifo][ids[ifo]]
+    c = rank_method.coinc_multiifo(sdswithids, slide, args.timeslide_interval)
+
+    #index values of the zerolag triggers
+    fi = numpy.where(slide == 0)[0]
+
+    #index values of the background triggers
+    bi = numpy.where(slide != 0)[0]
+    logging.info('%s foreground triggers' % len(fi))
+    logging.info('%s background triggers' % len(bi))
+
+    # We split the background triggers into two types which we keep track of
+    # in "bh" (triggers which are *not* decimated, stored in full) and
+    # "bl" (triggers which may not be stored in full, but we keep track of
+    # how many are removed)
+    # "bl_int" keeps track of which triggers are *not* in the bh set. Depending
+    # on the decimation factor option we may not store any of these or we may
+    # keep a fraction of them corresponding to a subset of the timeslides
+    if args.loudest_keep:
+        sep = len(bi) - args.loudest_keep
+        sep = 0 if sep < 0 else sep
+
+        bsort = numpy.argpartition(c[bi], sep)
+        bl_int = bi[bsort[0:sep]]
+        bh = bi[bsort[sep:]]
+        del bsort
+        del bi
+    elif args.loudest_keep_value:
+        bh = bi[c[bi] > args.loudest_keep_value]
+        bl_int = bi[c[bi] <= args.loudest_keep_value]
+    else:
+        bh = bi
+
+    if args.decimation_factor:
+        bl = bl_int[slide[bl_int] % args.decimation_factor == 0]
+    else:
+        bl = []
+
+    ti = numpy.concatenate([bl, bh, fi]).astype(numpy.uint32)
+    logging.info('%s after decimation' % len(ti))
+
+    gs={}
+    for ifo in ids:
+        gs[ifo]=ids[ifo][ti]
+    del ids
+
+    # In the first for cycle the time and trigger_id keys will not exist,
+    # we need to create them, otherwise add new items into data.
+    checkstring = '%s/time' % ReadByTemplate(args.trigger_files[0]).ifo
+    if checkstring in data:
+        for ifo in gs:
+            addtime=times[ifo][gs[ifo]]
+            addtriggerid=tids[ifo][gs[ifo]]
+            data['%s/time' % ifo] += [addtime]
+            data['%s/trigger_id' % ifo] += [addtriggerid]
+    else:
+        for ifo in gs:
+            addtime=times[ifo][gs[ifo]]
+            addtriggerid=tids[ifo][gs[ifo]]
+            data['%s/time' % ifo] = [addtime]
+            data['%s/trigger_id' % ifo] = [addtriggerid]
+    data['stat'] += [c[ti]]
+    dec_fac = numpy.repeat([args.decimation_factor, 1, 1],
+                           [len(bl), len(bh), len(fi)]).astype(numpy.uint32)
+    data['decimation_factor'] += [dec_fac]
+    data['timeslide_id'] += [slide[ti]]
+    data['template_id'] += [numpy.zeros(len(ti), dtype=numpy.uint32) + tnum]
+
+if len(data['stat']) > 0:
+    for key in data:
+        data[key] = numpy.concatenate(data[key])
+
+if args.cluster_window and len(data['stat']) > 0:
+    timestring0 = '%s/time' % ReadByTemplate(args.trigger_files[0]).ifo
```

Comment:
Shouldn't assume files are ordered. You can get the ifo names from the command line (see above comment)